tests/unit/tools/test_exception_run.py

```diff
@@ -0,0 +1,117 @@
+from unittest.mock import patch
+
+import pytest
+from litellm.types.utils import (
+    Choices,
+    Delta,
+    Function,
+    Message,
+    ModelResponse,
+    StreamingChoices,
+)
+
+from any_agent import (
+    AgentConfig,
+    AgentFramework,
+    AgentRunError,
+    AnyAgent,
+)
+from any_agent.tracing.otel_types import StatusCode
+
+
+def search_web(query: str) -> str:
+    """Perform a duckduckgo web search based on your query then returns the top search results.
+
+    Args:
+        query (str): The search query to perform.
+
+    Returns:
+        The top search results.
+
+    """
+    msg = "It's a trap!"
+    raise ValueError(msg)
+
+
+def test_tool_error_llm_mocked(
+    agent_framework: AgentFramework,
+) -> None:
+    """An exception raised inside a tool will be caught by us.
+
+    We make sure an appropriate Status is set to the tool execution span.
+    We allow the Agent to try to recover from the tool calling failure.
+    """
+
+    skip_reason = {
+        AgentFramework.AGNO: "Does not decide that the loop should end, and calls the tool the maximum amount of times before giving up",
+        AgentFramework.GOOGLE: "When mocking the LLM tool call, the tool calling does not happen and the loop is ended",
```

Comment:
When using the Google fw, somehow the loop silenty ends without indication of a tool failure (I'll try to debug this thoroughly)

Reply:
The integration test is not skipped and works for GOOGLE: https://github.com/mozilla-ai/any-agent/blob/22d2b750fde03768c658c5c67fe676fd8682a5a9/tests/integration/test_error_handling.py#L78 would be good to understand what is different here

Reply:
Apparently it's the same case for the Google framework.